### DISCUSSION TITLE
How to link attachments?

### DISCUSSION THREAD

**Discussion by zomaho** [TriliumNext/Trilium#4385] [Q&A]
Hello,

I don't understand how to link attachments to notes they belong to in version 0.61.

Case 1: I upload a video as attachment, I can see it when I select "Note attachments" in the menu. But how can I add link to the attachment in the note text, so that I can just click on it and see the video without going to "Note attachments"?

Case 2: I upload an image, via paragraph menu, it is added as attachment and appears in the note text. Then I delete image from note text, but it remains as attachment for some period of time. How can I add (link) it back to the note text? The paragraph menu only allows to insert image by uploading it again.

Thank you!

**Comment by meichthys** [comment]
In the web version you can just right click on the image and choose "Copy Image Link" (in Firefox). A similar option should be available in other browsers.

On desktop it doesn't seem possible. If you double click the image, then open the ... menu there is an option to copy the image link, but it actually copies the entire image to the clipboard (https://github.com/zadam/trilium/issues/4388)

**Reply by zomaho** [reply]
Well, this way I link to absolute URL which will break if I move to another host or port. Isn't there a way to link to attachment using relative internal URL,like we link to other notes?

**Comment by zadam** [comment]
Hi, you can use this:

![image](https://github.com/zadam/trilium/assets/617641/4e5dc43a-8912-4e40-ac71-d8175461a791)

**Comment by meichthys** [comment]
> Hi, you can use this:
> 
> ![image](https://github.com/zadam/trilium/assets/617641/4e5dc43a-8912-4e40-ac71-d8175461a791)
> 

Does this work in the latest? For me that button just copies the entire image to the clipboard, not the link to the image.

**Reply by zerebos** [reply]
It looks like latest does indeed copy the image itself

https://github.com/zadam/trilium/blob/stable/src/public/app/widgets/attachment_detail.js#L177 -> https://github.com/zadam/trilium/blob/master/src/public/app/services/image.js#L3

**Reply by zadam** [reply]
It's not easy to figure out what exactly happens since this is controlled by the platform (electron, browser), but:

```js
console.log(await (await (await navigator.clipboard.read())[0].getType("text/html")).text());
```

gives:

```html
<img src="http://127.0.0.1:37740/api/attachments/75HvFcwSDPaS/image/trilium-icon.png?2023-11-04%2022:39:47.873Z%22%3E" id="attachment-image-1" style="box-sizing: border-box; vertical-align: middle; border-style: none; margin: 10px; max-width: 100%; object-fit: contain; color: rgb(204, 204, 204); font-family: Montserrat; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(51, 51, 51); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; transition: transform 0.25s ease 0s; transform: translate(0px, 0px) scale(2.73438);">
```

So it is just link to the image.
